### PR TITLE
Fix crash when alias is referenced in an expression.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/67989

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@jcouv @333fred ptal.

**Review by arkalyanms — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler @jcouv ptal.

File: src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs [hunk 1]
```diff
@@ -5257,4 +5258,143 @@ public void UsingStatic_UnsafeType_CSharp12_UnsafeFlag()
         }
         EOF();
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/67989")]
+    public void AliasToPointerReferencedInExpression()
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs:5263]: I would like to see a test in the semantic layer that compiles successfully.

Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs:5263]: can you give an example of what you mean?

Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs:5263]: I do not think this test belongs in parsing tests. There is nothing unique about this scenario from the parsing perspective. The test should be moved to one of the binding related assemblies. For example, we have a ```UsingAliasTests``` type in Symbol tests.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs:5263]: > can you give an example of what you mean?  

```nameof(Alias)``` comes to mind, for example.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs:5263]: Assuming, it goes through the affected code path.

**Comment by AlekseyTs** [comment]
Done with review pass (commit 2)

**Review by 333fred — APPROVED** [review]
LGTM, assuming test is moved to symbol/semantic/emit layer.

File: src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs:5263]: I was previously asked to try to keep all the tests for a particular feature in single location.  Maybe we can change this from UsingDirectiveParsingTests, to just UsingDirectiveTests?

**Comment by CyrusNajmabadi** [comment]
Tests moved.

File: src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs]: > I was previously asked to try to keep all the tests for a particular feature in single location. 

Could you provide a link? Perhaps there was some confusion, we do not mix parsing tree tests with semantic tests, and. I believe, that started happening with changes that moved error reporting between layers in order to keep verifying diagnostics in the original tests. An exception like this is not applicable to new tests, especially those that have nothing to do with anything parsing related.

File: src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
 using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
 using Xunit;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs:8]: It looks like there are no meaningful change in this file now, please revert to the original state

**Comment by AlekseyTs** [comment]
Done with review pass (commit 4)

File: src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
 using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs:7]:
```suggestion
```

**Review by AlekseyTs — APPROVED** [review]
Changes under Compilers LGTM (commit 7)

**Pull request merged**